File: DoitMission_12/Doit12_Java/app/src/main/java/com/example/doit12_java/TextService.java
package com.example.doit12_java;

import android.app.Service;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.IBinder;
import android.util.Log;

public class TextService extends Service {

    public TextService() {
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        String name = intent.getStringExtra(MainActivity.name);
        String message = intent.getStringExtra(MainActivity.message);
        Intent broadIntent = new Intent();
        broadIntent.setAction("BroadCast");
        broadIntent.putExtra(MainActivity.name, name);
        broadIntent.putExtra(MainActivity.message, message);
        sendBroadcast(broadIntent);
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public IBinder onBind(Intent intent) {
        throw new UnsupportedOperationException("Not yet implemented");
    }
}

File: DoitMission_12/Doit12_Kotlin/app/src/main/java/com/example/doit12_kotlin/TextService.kt
package com.example.doit12_kotlin

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log

class TextService : Service() {

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        val message = intent.getStringExtra(MainActivity.message)
        val intentReceiver = Intent()
        intentReceiver.action = "BroadCast"
        intentReceiver.putExtra(MainActivity.message, message)
        Log.e("Start","서비스 onStartComman()")
        sendBroadcast(intentReceiver)
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder {
        throw UnsupportedOperationException("Not yet implemented")
    }
}
